### ISSUE TITLE
Nz-timepicker does not allow typed in options to stick and has no (keyup.enter) functionality. Also setting any step to interval will cause issues with times that are divisible by that interval.

### ISSUE THREAD

**Issue by ajhernandez95** [NG-ZORRO/ng-zorro-antd#6194]
### Reproduction link
[https://github.com/ajhernandez95/NG-ZORRO-BUG-SANDBOX](https://github.com/ajhernandez95/NG-ZORRO-BUG-SANDBOX)

### Steps to reproduce
Attempt to type in the time and hit enter to see if that time is actually sticking.

Next, type in the time and once the values are highlighted in the options list, try and click on the times and see that it does nothing once the times are highlighted due to the user typing in the time so no value is ever selected.

Finally, we had an issue where we had predefined times and nzMinuteStep set to 15 and unless the time was divisible by 15 it would not display correctly and it would round up/down to the closest value that is divisible by 15.

### What is expected?
I expected the value to still stick much like google's time picker. Even though their options can be set to interval you can still type in a custom value and that value will stick where as NG-ZORRO will not let the value stick unless its an option in the drop down.

Also we would like users to be able to type in the time that they would like and have that time automatically update the ngModel especially on (keyup.enter)

### What is actually happening?
Once you type in the time and hit enter nothing happens although it looks like you have the selected time you wanted.
Once options in the dropdown are highlighted due to the user typing in the time they become unselectable.

| Environment | Info |
|---|---|
| ng-zorro-antd | 9.3.0 |
| Browser | Chrome |

**Comment by JBetts97** [comment]
Hello @ajhernandez95,

Just to clarify with the "issue 1" in your replication link, do you mean that the TimePicker component stays visible after pressing enter, and the expected output is that the input is updated and the component preview window should be hidden (like it does when you click away from it)?

**Comment by ajhernandez95** [comment]
> Hello @ajhernandez95,
> Just to clarify with the "issue 1" in your replication link, do you mean that the TimePicker component stays visible after pressing enter, and the expected output is that the input is updated and the component preview window should be hidden (like it does when you click away from it)?

Correct, we expected that when you type in a value and hit enter it should select the value from the dropdown and close it. Did you need further clarification on the second or third issue?

**Comment by ajhernandez95** [comment]
@JBetts97 Checking in to see if their have been any updates.

**Comment by ajhernandez95** [comment]
@wenqi73 Could I get a status update on this issue?

**Comment by wenqi73** [comment]
@ajhernandez95 It will be fixed in the next version, sorry for the late reply.

**Comment by ajhernandez95** [comment]
> @ajhernandez95 `(keyup.enter)` has been supported, the other problem will be fixed in the next version, sorry for the late reply.

@wenqi73 I actually listed three issues total. Would you mind telling me which specific issue would be resolved? 

The issue where you are only allowed to type in values that are listed in the dropdown and not any other time? For example if we had the timer set up with intervals of 15 and we typed in 3:22 this does not work currently.

Or the issue where when you type in a value that is in the dropdown that value is no longer selectable?

Both these issues are pretty major for our organization.

And when you say the next version do you mean major version or patch version? Will we need to upgrade our complete front-end stack in order to resolve this issue? We currently use Angular 9 and NGZORRO 9.

**Comment by wenqi73** [comment]
> Or the issue where when you type in a value that is in the dropdown that value is no longer selectable?
> 
> I would also like to mention that although (keyup.enter) will trigger the event, the issue is more that the model is not changed when typing so even after I type in a value like 4:15 when the original value was 4:10 then 4:10 would be the value of the ngModel not 4:15 because the model never changed even though the typed in value did.

@ajhernandez95 These two issues are fixed in 10.0.0, the time interval issue is not fixed yet, we plan to fix this issue in 11.1.1, at that time you need to upgrade your project.

**Comment by Aravind0303** [comment]
@wenqi73 san and @ajhernandez95 , Is there any option,To restrict the entry in the time-picker????

Note : Its not about disable, Just to restrict the manual entry instead user can select in option(I mean in the dropdown )...
